### ISSUE TITLE
OpenGL demos error

### ISSUE THREAD

**Issue by quackzone** [factor/factor#650]
OpenGL demos fail to run on Ubuntu 12.04 with Nvidia drivers, this affects both factor 0.95 and the git repository.
##### Error

Running the bunny or spheres demo results in the following error:

`Memory protection fault at address 7f7339f80920`
##### Callstack

```
(U) Quotation: [ set-namestack init-catchstack self quot>> call -> stop ]
(O) Word: update-ui-loop
(O) Word: update-ui
(O) Word: notify-queued
(O) Method: M\ world graft*
(O) Word: set-up-window
(O) Method: M\ bunny-world begin-world
(O) Word: <bunny-buffers>
(O) Word: <gl-buffer>
(O) Word: gen-gl-buffer
(O) Word: glGenBuffers
(O) Word: gl-function-pointer
(U) Quotation: [
    OBJ-CURRENT-THREAD special-object error-thread set-global
    current-continuation -> error-continuation set-global
    [ original-error set-global ] [ rethrow ] bi
]
```
##### glxinfo

```
OpenGL vendor string: NVIDIA Corporation
OpenGL renderer string: Quadro NVS 135M/PCIe/SSE2
OpenGL version string: 3.3.0 NVIDIA 295.40
OpenGL shading language version string: 3.30 NVIDIA via Cg compiler
```

**Comment by erg** [comment]
This might actually be fixed now. Speculatively closing, reopen if it's still broken for you.

**Comment by timor** [comment]
I am encountering the same problem for 0.98 as well a recent version from master.  At least I think it is the same problem, the behavior is not a memory protection fault, but an undefined openGL function error. This is a NixOS system, so something else might be the problem here, but the solution is the same as above.

In fact, I am wondering why in `opengl.gl.extensions` under unix `opengl.gl.gtk` is loaded instead of `opengl.gl.x11`?
https://github.com/factor/factor/blob/41db42a5714819ab271ba04fae547a226fd7e47d/basis/opengl/gl/extensions/extensions.factor#L10

**Comment by mrjbq7** [comment]
You mean this fixes it?

https://github.com/factor/factor/issues/650#issuecomment-8282045

Fascinating... ok!

**Comment by timor** [comment]
> You mean this fixes it?
> 
> [#650 (comment)](https://github.com/factor/factor/issues/650#issuecomment-8282045)
Yes, but I simply replaced the whole line in `opengl.gl.extesions`.

I don't know much about the interaction between Gtk/Gdk/OpenGL, but if the "normal" graphics commands in `opengl.gl` use the GL API directly with "libGL.so", it seems a bit strange that the commands defined in `opengl.gl.extensions` should take a detour through the Gdk API.

**Comment by timor** [comment]
Oh, by the way, contrary to the initial report, I don't have an Nvidia card.  This happens on a Thinkpad X250 with Intel HD Graphics 5500 GPU.
